tests/test_deformers.py

```diff
@@ -734,3 +767,142 @@ def test_base_transformer():
+#                                   pytest.mark.xfail(1.1, raises=ValueError),
+                                  pytest.param(1.1, pytest.raises(ValueError), marks=pytest.mark.xfail),
+#                                   pytest.mark.xfail([0.2, 1.0], raises=ValueError)])
+                                  pytest.param([0.2, 1.0], pytest.raises(ValueError), marks=pytest.mark.xfail)])
+def test_clipping(clip_limit, expectation, jam_fixture):
+
+    with expectation: 
+        D = muda.deformers.Clipping(clip_limit=clip_limit)
+
+    jam_orig = deepcopy(jam_fixture)
+
+    for jam_new in D.transform(jam_orig):
+        # Verify that the original jam reference hasn't changed
+        assert jam_new is not jam_fixture
+        __test_time(jam_orig, jam_fixture, 1.0)
+
+        # Verify that the state and history objects are intact
+        __test_deformer_history(D, jam_new.sandbox.muda.history[-1])
+
+        d_state = jam_new.sandbox.muda.history[-1]['state']
+        d_clip_limit = d_state['clip_limit']
+        if isinstance(clip_limit, list):
+            assert d_clip_limit in clip_limit
+        else:
+            assert d_clip_limit == clip_limit
+
+        # Verify clipping outcome
+        __test_audio(jam_orig, jam_new, d_clip_limit)
+
+    # Serialization test
+    D2 = muda.deserialize(muda.serialize(D))
+    __test_params(D, D2)
+
+
+# LinearClipping
+@pytest.mark.parametrize('lower, upper',
+                         [(0.3, 0.5), (0.1, 0.9),
+#                           pytest.mark.xfail((-0.1, 0.2), raises=ValueError),
+                          pytest.param(-0.1, 0.2, marks=pytest.mark.xfail),
+#                           pytest.mark.xfail((1.0, 1.2), raises=ValueError),
+                          pytest.param(1.0, 1.2, marks=pytest.mark.xfail),
+#                           pytest.mark.xfail((0.8, 0.6), raises=ValueError),
+                          pytest.param(0.8, 0.6, marks=pytest.mark.xfail),
+#                           pytest.mark.xfail((0.6, 1.0), raises=ValueError)])
+                          pytest.param(0.6, 1.0, marks=pytest.mark.xfail)])
+def test_linear_clipping(n_samples, lower, upper, jam_fixture):
+
+    D = muda.deformers.LinearClipping(n_samples=n_samples,
+                                               lower=lower,
+                                               upper=upper)
+
+    jam_orig = deepcopy(jam_fixture)
+
+    n_out = 0    
+    for jam_new in D.transform(jam_orig):
+        # Verify that the original jam reference hasn't changed
+        assert jam_new is not jam_fixture
+        __test_time(jam_orig, jam_fixture, 1.0)
+
+        # Verify that the state and history objects are intact
+        __test_deformer_history(D, jam_new.sandbox.muda.history[-1])
+
+        d_state = jam_new.sandbox.muda.history[-1]['state']
+        d_clip_limit = d_state['clip_limit']
+        assert lower <= d_clip_limit <= upper
+
+        # Verify clipping outcome
+        __test_audio(jam_orig, jam_new, d_clip_limit)
```

Comment:
can we rename this helper function?  Many things could be "test audio", but this is specifically testing for clipping.

Reply:
Is `__test_clipped_buffer` okay?

Reply:
Changed.